modules/spring-config/src/main/java/org/mule/runtime/config/internal/context/lazy/LazyMuleArtifactContext.java

```diff
@@ -319,91 +370,97 @@ public Optional<ComponentModelInitializerAdapter> getParentComponentModelInitial
             .initializeComponents(componentModelPredicate, applyStartPhase));
   }
 
-  private List<Object> createComponents(Optional<Predicate<ComponentAst>> predicateOptional, Optional<Location> locationOptional,
-                                        boolean applyStartPhase,
+  private void validateModel(ArtifactAst artifactAst) throws ConfigurationException {
+    if (dslDeclarationValidationEnabled) {
+      doValidateModel(artifactAst, v -> v.getClass().getAnnotation(IgnoreOnLazyInit.class) == null
+          || v.getClass().getAnnotation(IgnoreOnLazyInit.class).forceDslDeclarationValidation());
+    } else {
+      doValidateModel(artifactAst, v -> v.getClass().getAnnotation(IgnoreOnLazyInit.class) == null);
+    }
+  }
+
+  private void initializeComponentsFromParent(Optional<ComponentModelInitializerAdapter> parentComponentModelInitializerAdapter) {
+    if (parentComponentModelInitializerAdapter.isPresent()) {
+      parentComponentModelInitializerAdapter.get()
+          .initializeComponents(componentModel -> graph.getMissingDependencies()
+              .stream()
+              .anyMatch(missingDep -> missingDep.isSatisfiedBy(componentModel)));
+    } else {
+      graph.getMissingDependencies().stream().forEach(missingDep -> {
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug("Ignoring dependency {} because it does not exist.", missingDep);
+        }
+      });
+    }
+  }
+
+  private void validateRequestedComponentExists(Location location, ArtifactAst minimalArtifactAst) {
+    if (minimalArtifactAst.recursiveStream()
+        .noneMatch(comp -> comp.getLocation() != null
+            && comp.getLocation().getLocation().equals(location.toString()))) {
+      throw new NoSuchComponentModelException(createStaticMessage("No object found at location "
+          + location.toString()));
+    }
+  }
+
+  private void cleanupAndResetComponentsState() {
+    // First unregister any already initialized/started component
+    unregisterBeans(currentComponentInitializationState.getTrackedBeansInOrder());
+
+    objectProviders.clear();
+    resetMuleSecurityManager();
+
+    // This has to be called after all previous state has been cleared because the unregister/cleanup process requires the
+    // errorTypeRepository as it was during its initialization.
+    // TODO: check this, it used to be done with the filtered AST, doing it like this might have a perf impact (not confirmed)
```

Comment:
I'm currently thinking if there would be an easy way to do this with a filtered AST. It is not straightforward because I'm not building the "accumulated" AST.
Maybe I'll leave it for a separate task. What I don't like about that is that it could be degrading performance of the current lazy initialization mechanism (though probably not by a lot, but it is complicated to measure without having representative scenarios).